mmaction/datasets/pose_dataset.py

```diff
@@ -0,0 +1,182 @@
+            For a video with n frames, it is a valid training sample only if
+            n * valid_ratio frames have human pose. None means not applicable
+            (only applicable to Kinetics Pose). Default: None.
+        box_thre (str | None): The threshold for human proposals. Only boxes
+            with confidence score larger than `box_thre` is kept. None means
+            not applicable (only applicable to Kinetics Pose [ours]). Allowed
+            choices are '0.5', '0.6', '0.7', '0.8', '0.9'. Default: None.
+        **kwargs: Keyword arguments for ``BaseDataset``.
+    """
+
+    def __init__(self,
+                 ann_file,
+                 pipeline,
+                 resample=None,
+                 valid_ratio=None,
+                 box_thre=None,
+                 **kwargs):
+        # For NTU-120 X-Sub
+        self.resample = resample
+        modality = 'Pose'
+
+        super().__init__(
+            ann_file, pipeline, start_index=0, modality=modality, **kwargs)
+
+        # box_thre, which should be a string
+        self.box_thre = box_thre
+        if self.box_thre is not None:
+            assert box_thre in ['0.5', '0.6', '0.7', '0.8', '0.9']
+
+        # Thresholding Training Examples
+        self.valid_ratio = valid_ratio
+        if self.valid_ratio is not None:
+            assert isinstance(self.valid_ratio, float)
+            if self.box_thre is None:
+                self.video_infos = self.video_infos = [
+                    x for x in self.video_infos
+                    if x['valid_frames'] / x['total_frames'] >= valid_ratio
+                ]
+            else:
+                key = f'valid@{self.box_thre}'
+                self.video_infos = [
+                    x for x in self.video_infos
+                    if x[key] / x['total_frames'] >= valid_ratio
+                ]
+                if self.box_thre != '0.5':
+                    box_thre = float(self.box_thre)
+                    for item in self.video_infos:
+                        inds = [
+                            i for i, score in enumerate(item['box_score'])
+                            if score >= box_thre
+                        ]
+                        item['anno_inds'] = np.array(inds)
+
+        logger = get_root_logger()
+        logger.info(f'{len(self)} videos remain after valid thresholding')
+
+    def load_annotations(self):
+        """Load annotation file to get video information."""
+        assert self.ann_file.endswith('.pkl')
+        return self.load_pkl_annotations()
+
+    def load_pkl_annotations(self):
+        data = mmcv.load(self.ann_file)
+
+        for i, item in enumerate(data):
+            # Sometimes we may need to load anno from the file
+            if 'filename' in item:
+                item['filename'] = osp.join(self.data_prefix, item['filename'])
+        return data
+
+    def prepare_train_frames(self, idx):
```

Comment:
These 3 functions can be removed since they are the same with those in BaseDataset

Reply:
I remove prepare_train_frames, prepare_test_frames but reserve the evaluate function. Since we only support top_k_accuracy and mean_class_accuracy for now. Besides, we may modify the evaluate function to support more features (like rgb+pose)